### PR TITLE
⬆️ Update penpal

### PR DESCRIPTION
**Pull request by JuaniRios**
## What?
- Update penpal runtime

## Why?
- To make sure our CT migration works with the latest xcm conventions

## How?
- Replace the old xcm with the new xcm from sdk 1.6.
- Add the polimec receiver and vesting pallets
- Modify the xcm config

## Testing?
`cargo test -p integration-tests`

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/322?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#322** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/322?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#317** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/317?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: Cargo.toml
Comment (vstam1) [inline Cargo.toml:1]: I would remove all the imports here that are only used for the penpal build and directly import them into them in the Cargo.toml of penpal. Now we need to build these pallets as well when compiling our full project, which in my eyes is just dependency bloat.

File: integration-tests/penpal/src/lib.rs [hunk 1]
```diff
@@ -128,7 +132,10 @@ pub type SignedExtra = (
 /// Unchecked extrinsic type as expected by this runtime.
 pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
 
-pub type Migrations = (pallet_balances::migration::MigrateToTrackInactive<Runtime, xcm_config::CheckingAccount>,);
+pub type Migrations = (
+	pallet_balances::migration::MigrateToTrackInactive<Runtime, xcm_config::CheckingAccount>,
+	pallet_collator_selection::migration::v1::MigrateToV1<Runtime>,
+);
```
Comment (vstam1) [inline integration-tests/penpal/src/lib.rs:138]: Why do we need a migration for this test runtime? Chain never launched, right?

File: integration-tests/penpal/src/lib.rs [hunk 2]
```diff
@@ -457,11 +463,43 @@ impl pallet_assets::Config for Runtime {
...
+	// we just reuse the same deposits
+	pub const ForeignAssetsAssetDeposit: Balance = AssetDeposit::get();
+	pub const ForeignAssetsAssetAccountDeposit: Balance = AssetAccountDeposit::get();
+	pub const ForeignAssetsApprovalDeposit: Balance = ApprovalDeposit::get();
+	pub const ForeignAssetsAssetsStringLimit: u32 = AssetsStringLimit::get();
+	pub const ForeignAssetsMetadataDepositBase: Balance = MetadataDepositBase::get();
+	pub const ForeignAssetsMetadataDepositPerByte: Balance = MetadataDepositPerByte::get();
+}
+
+/// Another pallet assets instance to store foreign assets from bridgehub.
```
Comment (vstam1) [inline integration-tests/penpal/src/lib.rs:475]: ? Why do we need foreign assets store for bridgehub assets?

File: integration-tests/penpal/src/xcm_config.rs [hunk 3]
```diff
@@ -93,12 +96,20 @@ pub type FungiblesTransactor = FungiblesAdapter<
...
+			AsPrefixedGeneralIndex<AssetsPalletLocation, AssetIdPalletAssets, JustTry>,
+			JustTry,
+		>,
+		ConvertedConcreteId<
+			AssetIdPalletAssets,
+			Balance,
+			AsPrefixedGeneralIndex<SystemAssetHubAssetsPalletLocation, AssetIdPalletAssets, JustTry>,
+			JustTry,
+		>,
+	),
```
Comment (vstam1) [inline integration-tests/penpal/src/xcm_config.rs:112]: Not Really needed in our testing env?

File: integration-tests/penpal/src/xcm_config.rs [hunk 4]
```diff
@@ -110,8 +121,28 @@ pub type FungiblesTransactor = FungiblesAdapter<
...
+	// Convert an XCM MultiLocation into a local account id:
+	LocationToAccountId,
+	// Our chain's account ID type (we can't get away without mentioning it explicitly):
+	AccountId,
+	// We dont need to check teleports here.
+	NoChecking,
+	// The account to use for tracking teleports.
+	CheckingAccount,
+>;
+
```
Comment (vstam1) [inline integration-tests/penpal/src/xcm_config.rs:143]: Can we skip this?

File: integration-tests/penpal/src/xcm_config.rs [hunk 5]
```diff
@@ -126,7 +157,10 @@ pub type XcmOriginToTransactDispatchOrigin = (
 	RelayChainAsNative<RelayChainOrigin, RuntimeOrigin>,
 	// Native converter for sibling Parachains; will convert to a `SiblingPara` origin when
 	// recognized.
-	SiblingParachainAsNative<polkadot_runtime_parachains::origin::Origin, RuntimeOrigin>,
+	SiblingParachainAsNative<cumulus_pallet_xcm::Origin, RuntimeOrigin>,
+	// Superuser converter for the Relay-chain (Parent) location. This will allow it to issue a
+	// transaction from the Root origin.
+	ParentAsSuperuser<RuntimeOrigin>,
```
Comment (vstam1) [inline integration-tests/penpal/src/xcm_config.rs:163]: Can probably skip this as well if we want

File: integration-tests/penpal/src/xcm_config.rs
Comment (vstam1) [inline integration-tests/penpal/src/xcm_config.rs:1]: This file contains quite a lot of configuration for working with AssetHub. Do we actually need it for testing direct migrations? Or am I missing something for our integration tests? Else its just bloat/complicated code, that might even give unexpected results.

File: runtimes/polimec/src/xcm_config.rs [hunk 6]
```diff
@@ -226,7 +226,37 @@ match_types! {
 		MultiLocation { parents: 1, interior: X1(Parachain(_)) }
 	};
 }
-
+use polimec_xcm_executor::polimec_traits::OnResponse;
+/// Allows only messages if the generic `ResponseHandler` expects them via `expecting_response`.
+pub struct AllowKnownQueryResponses<ResponseHandler>(PhantomData<ResponseHandler>);
```
Comment (vstam1) [inline runtimes/polimec/src/xcm_config.rs:231]: Why do we need this custom struct? Is the xcm-builder type not correct?

File: integration-tests/penpal/src/lib.rs (hunk 2 above)
Comment (JuaniRios) [reply integration-tests/penpal/src/lib.rs:475]: It's how penpal comes out of the box. I prefer to leave it as is

File: integration-tests/penpal/src/xcm_config.rs (hunk 3 above)
Comment (JuaniRios) [reply integration-tests/penpal/src/xcm_config.rs:112]: Not needed, but we choose keep the runtime as close as the original as possible

File: runtimes/polimec/src/xcm_config.rs (hunk 6 above)
Comment (JuaniRios) [reply runtimes/polimec/src/xcm_config.rs:231]: It's bugged. I'll submit an issue soon.
In short, they made XCM router by default to append a `SetTopic` at the end of every sent message, and so a query response will contain 2 messages. The default `AllowKnownQueryResponses` expects only 1.

**Review by lrazovic — APPROVED** [review]
If the conversation around the `AllowKnownQueryResponses` is resolved (@vstam1), we can merge.

**Pull request merged**